src/librustc_driver/args/mod.rs

```diff
@@ -0,0 +1,207 @@
+use std::env;
+use std::ffi::OsString;
+use std::fs;
+use std::mem;
+use std::path::PathBuf;
+
+#[cfg(test)]
+mod tests;
+
+// Max number of recursive @file includes
+const MAX_RECURSION: usize = 8;
```

Comment:
I never really full understood the purpose of recursion myself. It always seemed like these types of argument inputs are basically never used except tooling calling tooling. Apart from the existing convention, do we have a real-world use case of supporting recursion?

Reply:
I can certainly make some things up which sound semi-plausible, but I don't actually have a concrete use case.

The recursive `@some/path` are a bit awkward to my mind - the paths are always relative to the cwd rather than the including file, as is the convention for other kinds of includes. I just copied the gnu-ld and gcc behaviour (determined empirically since it isn't documented)